src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -500,7 +518,8 @@ private bool ReplacementChangesSemanticsForNode(TSyntaxNode currentOriginalNode,
             {
                 var originalExpression = (TExpressionSyntax)currentOriginalNode;
                 var newExpression = (TExpressionSyntax)currentReplacedNode;
-                if (!ImplicitConversionsAreCompatible(originalExpression, newExpression))
+                if (!ImplicitConversionsAreCompatible(originalExpression, newExpression) ||
+                    ReplacementIntroducesErrorType(originalExpression, newExpression))
                 {
                     return true;
```

Comment:
I don't know this code that much but, TypesAreCompatible returning true when it introduced an error feels weird.